lib/tools/stats.js

```diff
@@ -0,0 +1,17 @@
+var convert = require('./convert_time');
+var BigInt = require('./bigint');
+
+module.exports = function(file) {
+  return {
+    birthtime: convert(BigInt.fromBuffer(file.CreationTime).toNumber()),
+    atime: convert(BigInt.fromBuffer(file.LastAccessTime).toNumber()),
+    mtime: convert(BigInt.fromBuffer(file.LastWriteTime).toNumber()),
+    ctime: convert(BigInt.fromBuffer(file.ChangeTime).toNumber()),
+    size: BigInt.fromBuffer(file.EndofFile).toNumber(),
+    isDirectory: function() {
+      if (typeof file.FileAttributes === 'number')
+        return file.FileAttributes === 16;
+      return BigInt.fromBuffer(file.FileAttributes).toNumber() === 16;
```

Comment:
Why are they two cases?

Reply:
by running the tests, I saw that sometimes isDirectory was not returning proper result. After checking the raw data, I saw that sometimes the FileAttributes attribute was sometimes an integer and some others a buffer.